tests/unit/libbpf_test.cpp

```diff
@@ -3429,10 +3459,13 @@ _test_maps_batch(bpf_map_type map_type)
 
     uint32_t batch_size = 20000;
     std::vector<uint32_t> keys(batch_size);
-    std::vector<uint64_t> values(batch_size);
+    std::vector<uint64_t> values(batch_size * value_size);
```

Comment:
consider
```
using per_cpu_map_value = vector<uint64_t>;
using batch_of_values = vector<per_cpu_map_values>;

batch_of_values values(batch_count, per_cpu_map_value(num_cpu_count));

for (i = 0; i < batch_count; i++) {
   for (cpu = 0; cpu < num_cput_count; cpu++) {
      values[i][cpu] = 2*i;
   }
}
````

Reply:
The above snippet is a vector of vector. 
It is better to keep it simple as a vector (instead of vector of vector), where it suffices for both non-per-cpu (num of CPU is 1) and per-cpu.

Reply:
As discussed offline with you regarding the comment of considering the approach of vector of vector, the buffer/values need to be flattened again to provide to ebpf update batch API.